qcodes/instrument_drivers/Keysight/Keysight_81180A.py

```diff
@@ -53,360 +76,399 @@ def __init__(self, parent, name, id, **kwargs):
-                      'the AC output. DC and DAC couplings also allow '
-                      'control of amplitude and offset')
-
-        self.add_parameter(
-            'output',
-            get_cmd='OUTPUT?',
-            set_cmd='OUTPUT {}',
-            vals=EnumVisa('ON', 'OFF'))
-
-        self.add_parameter(
-            'sync',
-            get_cmd='OUTPUT:SYNC?',
-            set_cmd='OUTPUT:SYNC {}',
-            vals=EnumVisa('ON', 'OFF')
-        )
-
-        self.add_parameter(
-            'power',
-            unit='dBm',
-            get_cmd='POWER?',
-            set_cmd='POWER {:.8f}',
+            docstring="Set the frequency of standard waveforms",
+        )
+
+        self.add_parameter(
+            "function_mode",
+            get_cmd="FUNCTION:MODE?",
+            set_cmd="FUNCTION:MODE {}",
+            vals=EnumCaseInsensitive(
+                "FIXed", "USER", "SEQuenced", "ASEQuenced", "MODulated", "PULSe"
+            ),
+            docstring="Run mode defines the type of waveform that is "
+            "available. Possible run modes are:\n"
+            "\tfixed: standard waveform shapes.\n"
+            "\tuser: arbitrary waveform shapes.\n"
+            "\tsequenced: sequence of arbitrary waveforms.\n"
+            "\tasequenced: advanced sequence of arbitrary "
+            "waveforms.\n"
+            "\tmodulated: modulated (standard) waveforms.\n"
+            "\tpulse: digital pulse function",
+        )
+
+        self.add_parameter(
+            "output_coupling",
+            get_cmd="OUTPUT:COUPLING?",
+            set_cmd="OUTPUT:COUPLING {}",
+            vals=vals.Enum("DC", "DAC", "AC"),
+            docstring="Possible output couplings are:\n"
+            "\tDC: optimized for pulses at high amplitudes.\n"
+            "\tDAC: optimized for bandwidth but low amplitude.\n"
+            "\tAC: optimized for bandwidth.\n"
+            "Note that DC and DAC use the DC output, and AC uses "
+            "the AC output. DC and DAC couplings also allow "
+            "control of amplitude and offset",
+        )
+
+        self.add_parameter(
+            "output",
+            get_cmd="OUTPUT?",
+            set_cmd="OUTPUT {}",
+            vals=EnumCaseInsensitive("ON", "OFF"),
+        )
+
+        self.add_parameter(
+            "sync",
+            get_cmd="OUTPUT:SYNC?",
+            set_cmd="OUTPUT:SYNC {}",
+            vals=EnumCaseInsensitive("ON", "OFF"),
+        )
+
+        self.add_parameter(
+            "power",
+            unit="dBm",
+            get_cmd="POWER?",
+            set_cmd="POWER {:.8f}",
             get_parser=float,
-            vals=vals.Numbers(-8, 8), # Might be -5 to 5, manual is unclear
-            docstring='Set output power from AC output path (50-ohm matched)')
+            vals=vals.Numbers(-5, 5),  # Might be -5 to 5, manual is unclear
+            docstring="Set output power from AC output path (50-ohm matched)",
```

Comment:
Write unit in docstring for readability.

Reply:
added

Reply:
done